packages/shorebird_cli/lib/src/executables/xcodebuild.dart

```diff
@@ -106,4 +107,41 @@ class XcodeBuild {
       schemes: schemes,
     );
   }
+
+  /// Gets the currently installed version of Xcode.
+  ///
+  /// Invokes `xcodebuild -version` and parses the output.
+  /// Output is expected to be of the form:
+  ///
+  ///   $ /usr/bin/xcodebuild -version
+  ///   Xcode 15.0
+  ///   Build version 15A240d
+  Future<Version> xcodeVersion() async {
+    const arguments = ['-version'];
+    final result = await process.run(
+      executable,
+      arguments,
+    );
+
+    if (result.exitCode != ExitCode.success.code) {
+      throw ProcessException(executable, arguments, '${result.stderr}');
+    }
+
+    final lines = LineSplitter.split('${result.stdout}').map((e) => e.trim());
+    var versionString = lines.firstOrNull?.split(' ').lastOrNull;
+    if (versionString == null) {
+      throw FormatException(
+        'Could not parse Xcode version from output: "${result.stdout}".',
+      );
+    }
+
+    // [Version.parse] requires a patch number. If Xcode does not report a patch
```

Comment:
I wonder if this will work for all XCode versions and/or whether it would be easier and less error-prone to try to parse the version as a double and see if it’s greater than 15 since clearly XCode doesn’t follow semver I’m not sure this would be the only issue we encounter (for example if there’s a beta version of XCode do we handle that correctly with this logic). 🤷‍♂️

Reply:
I guess we should probably be consistent with the flutter_tool in this case

Reply:
Good Q about the beta, let me download one and give that a shot

Reply:
Interestingly, Xcode betas don't report that they're beta in `xcodebuild -version`:

```
⑆ xcrun xcodebuild -version
Xcode 15.1
Build version 15C5028h
```

Reply:
To resolve this, I don't think parsing it as a double would work, as there is occasionally a patch version.